02_bubble/jonathanheaden/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	done := false
+	if len(s) == 0 {
+		return s
+	}
+	for !done {
+		sw := false
+		for i := 0; i < len(s)-1; i++ {
+			if s[i] > s[i+1] {
+				sw = true
+				s[i], s[i+1] = s[i+1], s[i]
+			}
+			done = !sw
+		}
+	}
+	return s
```

Comment:
Does this mutate the original slice? If so is that what you intended?

Reply:
Yes it does mutate the original slice, and for the purpose of this exercise it is what I want. If I were implementing without already having the func signature defined in the readme I would not have it return anything but instead mutate the slice in place as written

Reply:
Actually, thinking more on this, I now agree it should not mutate the original slice. if the signature of the function didn't return anything then mutation would be expected. As the signature is written it should not mutate the passed in slice but work on a copy instead. I will fix this up